src/squlearn/encoding_circuit/qcnn_encoding_circuit.py

```diff
@@ -0,0 +1,686 @@
+import numpy as np
+from typing import Union
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector
+from qiskit.converters import circuit_to_gate, circuit_to_instruction
+
+from squlearn.encoding_circuit.encoding_circuit_base import EncodingCircuitBase
+from squlearn.encoding_circuit.circuit_library.param_z_feature_map import ParamZFeatureMap
+from squlearn.observables import CustomObservable
+
+
+class QCNNEncodingCircuit(EncodingCircuitBase):
+
+    """
+    Encoding circuit for quantum convolutional neural networks (QCNN).
+
+    The structure is inspired by classical neural networks.
+    Therefore the number of qubits it is operated on is reduced from layer to layer.
+    For more information read:
+    "Quantum convolutional neural networks" by Cong, Iris and Choi, Soonwon and Lukin, Mikhail D.
+
+    Args:
+        num_qubits (int): Number of initial qubits of the QCNN encoding circuit.
+        num_features (int): Dimension of the feature vector.
+            By default this is 0, so a feature map must be provided.
+            If the number of features is bigger then 0,
+            then in the get_circuit function a ZFeatureMap is built to encode the features.
+        default (bool): If True, the default circuit is built.
+    """
```

Comment:
A bit clearer and right way to reference
```suggestion
    The structure is inspired by classical convolutional neural networks. The number of active
    qubits reduces with each layer. The design idea was initially proposed in reference [1].

    Args:
        num_qubits (int): Number of initial qubits of the QCNN encoding circuit.
        num_features (int): Dimension of the feature vector.
            By default this is 0, so a feature map must be provided.
            If the number of features is bigger then 0,
            then in the get_circuit function a ZFeatureMap is built to encode the features.
        default (bool): If True, the default circuit is built.
    
    References
    -----------
    [1]: `Cong, I., Choi, S. & Lukin, M.D. Quantum convolutional neural networks. Nat. Phys. 15,
    1273–1278 (2019). <https://doi.org/10.1038/s41567-019-0648-8>`_
    """
```

Reply:
The reference section in the bottom is still missing